roofit/roofitcore/src/RooWorkspace.cxx

```diff
@@ -2212,11 +2212,11 @@ Bool_t RooWorkspace::writeToFile(const char* fileName, Bool_t recreate)
 RooFactoryWSTool& RooWorkspace::factory()
 {
   if (_factory) {
-    return *_factory ;  
+    return *_factory.get();  
   }
   cxcoutD(ObjectHandling) << "INFO: Creating RooFactoryWSTool associated with this workspace" << endl ;
-  _factory = new RooFactoryWSTool(*this) ;
-  return *_factory  ;
+  _factory.reset(new RooFactoryWSTool(*this));
```

Comment:
We have a backport of make_unique in order to use the construct in C++11 as well: perhaps this is a nice place where we can use it?
The header inclusion to take advantage of this construct is:
```
#include "ROOT/RMakeUnique.hxx"
```